### PR TITLE
add clustering example

### PR DESCRIPTION
**Pull request by yukimori**
related to #31

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,23 +69,28 @@ Jubatus Examples
 	- classifier
 	- Python
 	- 画像特徴抽出プラグインを用いた簡単なサンプルです。mnistデータセットを利用して、画像から数字を推定します。
+18. clustering
+	- clustering
+	- Python
+	- 2次元座標をもつデータを使って、クラスタリングを行います
```
Comment (yukimori) [inline README.md]: I fix it.

**Comment by yukimori** [comment]
I think Jubatus-example doesn't have module dependencies which are like matplotlib or numpy.
But I understand it is difficult to see how jubatus cluster, So I modify to save result after checking module dependencies.

File: clustering/clustering.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+try:
+    import matplotlib.pyplot as plt
+    import numpy as np
+    isVisualization = True
+    colors = np.array([x for x in 'bgrcmykbgrcmykbgrcmykbgrcmyk'])
+    colors = np.hstack([colors] * 20)
+    plt.figure(figsize=(len(datasets_name) * 2 + 6, 4))
+except ImportError:
+    print("no matplotlib or numpy, so can't visualize clustering")
+    isVisualization = False
```
Comment (shiodat) [inline clustering/clustering.py]: CamelCase and snake_case are mixed. 
Should we write `is_visualization` or simply `visualization`?

File: clustering/clustering.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+plot_num = 1
+for i_dataset, (name, dataset) in enumerate(datasets.items()):
+    client.clear()
+
+    if isVisualization:
+        plt.title(datasets_name[i_dataset], size=18)
```
Comment (shiodat) [inline clustering/clustering.py]: Title plotting must be done after calling `plt.subplot(...)` at line 57.
Please move this line to line 58.

File: clustering/clustering.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    print("start to push {0} data with id(execute clustering {1} times)".format(name, client.get_revision()))
+    for i, data in enumerate(dataset):
+        client.push([( str(i) , Datum({"x" : data['x'], "y" : data['y']}))])
+    print("{0} data pushed(calculate cluster : {1} times)".format(len(dataset), client.get_revision()))
+    
+    clusters = client.get_core_members_light()
+    print("get cluster info. it has {0} clusters".format(len(clusters)))
+    for i, cluster in enumerate(clusters):
+        if isVisualization:
+            plt.subplot(1, len(datasets_name), plot_num)
```
Comment (shiodat) [inline clustering/clustering.py]: `i_dataset` and `plot_num` seem redundant for me. 
`plot_num` is used only once, and I suggest `plt.subplot(1, len(datasets_name), i_dataset+1)`.

File: clustering/clustering.py (hunk 3 above)
Comment (yukimori) [inline clustering/clustering.py]: move to line 58

File: clustering/clustering.py (hunk 4 above)
Comment (yukimori) [inline clustering/clustering.py]: I fix it

**Comment by yukimori** [comment]
Thank you for your comment. I fix them.

File: clustering/README.rst [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+以下のようにプログラムを実行してください。
+
+::
+   
+ $ python clustering.py
+
+このサンプルコードではデータセットを読み込み、
+jubaclusteringにデータを追加してクラスタリングを実施し、
+クラスタリング結果を取得して、出力しています。
+前述したグラフを出力するライブラリが事前にインストールされている場合は
```
Comment (shiodat) [inline clustering/README.rst]: Unfinished description? Please read the whole sentences and rewrite them.

Comment (yukimori) [inline clustering/README.rst]: I fix t

File: clustering/clustering.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    for i, data in enumerate(dataset):
+        client.push([( str(i) , Datum({"x" : data['x'], "y" : data['y']}))])
+    print("{0} data pushed(calculate cluster : {1} times)".format(len(dataset), client.get_revision()))
+    
+    clusters = client.get_core_members_light()
+    print("get cluster info. it has {0} clusters".format(len(clusters)))
+    for i, cluster in enumerate(clusters):
+        if is_visualization:
+            plt.title(datasets_name[i_dataset], size=18)
+            plt.subplot(1, len(datasets_name), (i_dataset+1))
```
Comment (shiodat) [inline clustering/clustering.py]: `plt.title(...)` must be called AFTER plt.subplot(...).
Please swap line 53 and line 54.

**Comment by yukimori** [comment]
I fix the title

**Comment by shiodat** [comment]
:+1:

**Pull request merged**